packages/js/product-editor/src/blocks/text/edit.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { createElement, createInterpolateElement } from '@wordpress/element';
+import { useBlockProps } from '@wordpress/block-editor';
+import { useInstanceId } from '@wordpress/compose';
+import { __, sprintf } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
+import classNames from 'classnames';
+import {
+	BaseControl,
+	// @ts-expect-error `__experimentalInputControl` does exist.
+	__experimentalInputControl as InputControl,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { useValidation } from '../../contexts/validation-context';
+import useProductEntityProp from '../../hooks/use-product-entity-prop';
+import { TextBlockAttributes } from './types';
+import { ProductEditorBlockEditProps } from '../../types';
+
+export function Edit( {
+	attributes,
+}: ProductEditorBlockEditProps< TextBlockAttributes > ) {
+	const blockProps = useBlockProps();
```

Comment:
We should use `useWooBlockProps()` in all of our blocks, since that is part of the magic that auto-adds the template block ID and order to the the DOM to allow for tools like the inspector to work.